airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -273,6 +279,14 @@ def _get_default_client(self, *, cluster_context=None):
             )
         return client.ApiClient()
 
+    @property
+    def is_in_cluster(self):
+        """Expose whether the hook is configured with incluster_config or not"""
+        if self._is_in_cluster is not None:
+            return self._is_in_cluster
+        self.get_conn()  # so we can determine if we are in_cluster or not
```

Comment:
since get_conn isn't cached it seems better to call self.api_client

Reply:
Works for me.